gcloud/pubsub/client.py

```diff
@@ -65,20 +65,12 @@ def list_topics(self, page_size=None, page_token=None):
                   more topics can be retrieved with another call (pass that
                   value as ``page_token``).
         """
-        params = {}
-
-        if page_size is not None:
-            params['pageSize'] = page_size
-
-        if page_token is not None:
-            params['pageToken'] = page_token
-
-        path = '/projects/%s/topics' % (self.project,)
-        resp = self.connection.api_request(method='GET', path=path,
-                                           query_params=params)
+        conn = self.connection
+        resources, next_token = conn.list_topics(
```

Comment:
@tseaver This doesn't seem to go all the way to being able to handle JSON or gRPC. ISTM that `resources` will still be parsed JSON but a gRPC API would return a protobuf object. Do you expect to retro-fit the `from_api_repr` methods to accept a `dict` or a protobuf or do you expect to fake the dict by deserializing the protobuf to a dict first?